contracts/vc/VVVVCReadOnlyInvestmentLedger.sol

```diff
@@ -0,0 +1,133 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.23;
+
+import { AccessControl } from "@openzeppelin/contracts/access/AccessControl.sol";
+import { ECDSA } from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+
+contract VVVVCReadOnlyInvestmentLedger is AccessControl {
+    /// @notice EIP-712 standard definitions
+    bytes32 public constant DOMAIN_TYPEHASH =
+        keccak256(bytes("EIP712Domain(string name,uint256 chainId,address verifyingContract)"));
+    bytes32 public constant SET_ROOTS_TYPEHASH =
+        keccak256(
+            bytes(
+                "RoundStateSet(uint256 investmentRound, bytes32 kycAddressInvestedRoot, bytes32 totalInvestedRoot, uint256 timestamp, uint256 roundNonce, uint256 totalNonce)"
+            )
+        );
+    bytes32 public immutable DOMAIN_SEPARATOR;
+
+    bytes32 public constant ROOT_SETTER_ROLE = keccak256("ROOT_SETTER_ROLE");
+
+    /// @notice The address authorized to sign investment transactions
+    address public signer;
+
+    /// @notice nonce for all combined round state updates
+    uint256 public totalNonce;
+
+    /// @notice root for each round which validates kyc address amounts invested
+    mapping(uint256 => bytes32) public kycAddressInvestedRoots;
+
+    /// @notice root for each round which validates total amounts invested
+    mapping(uint256 => bytes32) public totalRoots;
```

Comment:
True good point, ok so for now, I'll run with a root + a value per round. It'd maybe be a bit of gas savings otherwise, but this is a bit more transparent.